### PR TITLE
Use direct switch for small IN lists of small primitives #3164

### PR DESCRIPTION
**Pull request by Myrthan**
https://github.com/facebook/presto/issues/3164

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java [hunk 1]
```diff
@@ -99,7 +101,11 @@ public ByteCodeNode generateExpression(Signature signature, ByteCodeGeneratorCon
         Scope scope = generatorContext.getScope();
 
         ByteCodeNode switchBlock;
-        if (constantValues.size() < 1000) {
+
+        if (maxValue <= Integer.MAX_VALUE && minValue >= Integer.MIN_VALUE) {
+            switchBlock = new Block().comment("value between max and min integer").dup(javaType).append(ByteCodeUtils.boxPrimitive(javaType)).longToInt();
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java]: This seems to be missing the actual `switch` code. Can you make sure this case is covered by the existing tests and if not, add a test?

File: presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java [hunk 2]
```diff
@@ -67,12 +64,17 @@ public ByteCodeNode generateExpression(Signature signature, ByteCodeGeneratorCon
...
+        long maxValue = -1;
+        long minValue = 1;
+
         for (RowExpression testValue : values) {
             ByteCodeNode testByteCode = generatorContext.generate(testValue);
 
             if (testValue instanceof ConstantExpression && ((ConstantExpression) testValue).getValue() != null) {
                 ConstantExpression constant = (ConstantExpression) testValue;
                 Object object = constant.getValue();
+                maxValue = Math.max((long) object, maxValue);
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java]: This cast will fail if the object isn't a `long`.

File: presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java [hunk 3]
```diff
@@ -30,10 +30,7 @@
 import com.google.common.collect.ImmutableListMultimap;
 import com.google.common.collect.ImmutableSet;
 
-import java.util.Collection;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (yuananf) [inline presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java]: This star import causes checkstyle error. 
Why is this needed?

Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java]: Our code style doesn't allow them because they cause problems (if you star import multiple packages, then later the same class is added to both packages, your code no longer compiles).

Your IDE should have an option to turn it off (or set the threshold very high). If you use IDEA we have a code style here: https://github.com/airlift/codestyle

**Comment by cberner** [comment]
@Myrthan are you still working on this?

**Comment by cberner** [comment]
Please re-open, if you're still working on this
